frontend/src/views/website/runtime/dir/index.vue

```diff
@@ -0,0 +1,136 @@
+                        {{ $t('runtime.runScriptHelper') }}
+                    </span>
+                    <span class="input-help" v-else>
+                        {{ scriptHelper }}
+                    </span>
+                </el-form-item>
+            </el-col>
+            <el-col :span="6">
+                <el-form-item :label="$t('runtime.customScript')" prop="params.CUSTOM_SCRIPT">
+                    <el-switch
+                        v-model="runtime.params['CUSTOM_SCRIPT']"
+                        :active-value="'1'"
+                        :inactive-value="'0'"
+                        @change="changeScriptType"
+                    />
+                </el-form-item>
+            </el-col>
+        </el-row>
+    </div>
+    <div v-else>
+        <el-form-item :label="$t('runtime.runScript')" prop="params.EXEC_SCRIPT">
+            <el-input v-model="runtime.params['EXEC_SCRIPT']"></el-input>
+            <span class="input-help">
+                {{ scriptHelper }}
+            </span>
+        </el-form-item>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { Runtime } from '@/api/interface/runtime';
+import { GetNodeScripts } from '@/api/modules/runtime';
+import { useVModel } from '@vueuse/core';
+import { defineProps } from 'vue';
+
+const props = defineProps({
+    mode: {
+        type: String,
+        required: true,
+    },
+    modelValue: {
+        type: Object,
+        required: true,
+    },
+    dirHelper: {
+        type: String,
+        required: false,
+    },
+    scriptHelper: {
+        type: String,
+        required: true,
+    },
+    appKey: {
+        type: String,
+        required: false,
+    },
+});
+const emit = defineEmits(['update:modelValue']);
+const runtime = useVModel(props, 'modelValue', emit);
+const scripts = ref<Runtime.NodeScripts[]>([]);
+
+watch(
+    () => runtime.value.name,
+    (newVal) => {
+        if (newVal && props.mode == 'create') {
+            runtime.value.params['CONTAINER_NAME'] = newVal;
+        }
+    },
+    { deep: true },
+);
+
+const changeDir = () => {
+    if (props.appKey == 'node') {
+        getScripts();
+    }
+};
+
+const getPath = (codeDir: string) => {
+    runtime.value.codeDir = codeDir;
+    if (props.appKey == 'node') {
+        getScripts();
+    }
+};
+
+const changeScriptType = () => {
+    runtime.value.params['EXEC_SCRIPT'] = '';
+    if (runtime.value.params['CUSTOM_SCRIPT'] == '0') {
+        getScripts();
+    }
+};
+
+const getScripts = () => {
+    GetNodeScripts({ codeDir: runtime.value.codeDir }).then((res) => {
+        scripts.value = res.data;
+        if (props.mode == 'create' && scripts.value.length > 0) {
+            runtime.value.params['EXEC_SCRIPT'] = scripts.value[0].name;
+        }
+    });
+};
+
+onMounted(() => {
+    if (props.mode == 'edit' && props.appKey == 'node') {
+        getScripts();
+    }
+});
+</script>
```

Comment:
The above code appears to be TypeScript-based Vue.js template code. Let's go over some of its features:

1. There is a `getScripts` function which makes an asynchronous request using `GET` method to fetch data from Node API for a given Code Directory.

2 The code is structured well with clear functions defined, including `defineProps()`, `useVModel()` and a recursive watch mechanism (`watch(...)`), which suggests that it uses ES6 classes under the hood.

3. However, there are no syntax errors found in the provided snippet based on available information.

There does not appear to be any obvious issues or bugs in this code as currently written.

No optimizations suggested due to brevity and clarity. If there were major improvements needed (e.g., more performance enhancements, changes to handle edge cases better, etc.), that would need more thorough review than I am able to do here.